### DISCUSSION TITLE
v20.0.0-beta.3

### DISCUSSION THREAD

**Discussion by jonyo** [geodesicsolutions-community/geocore-community#71] [Announcements]
## What's Changed since beta 2

* Fix the PHP version it shows when upgrading and the version check fai… by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/46
* Admin login re-skin for GeoCore CE by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/47
* Make links during setup include the port to work better in local dock… by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/48
* Fix SQL query error in storefront cron, trying to pass a query param … by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/51
* Going through results of cs-check and cleaning up a bunch of files by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/52
* Update ADODB to latest version, and switch to using composer by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/54
* Fix up lint issues on the previously encoded files, and remove some deprecated methods by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/69
* Vscode xdebug settings that work with docker by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/70

## What's Changed since 18.02.0
* Import legacy code - this imports the code with a fresh start from latest release by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/7
* Remove log license addon by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/8
* Apply all the un-released changes from legacy codeset by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/10
* Remove OS files and add them to .gitignore by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/11
* Update installation requirements check and links by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/14
* Add docker support by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/17
* Update the admin panel wiki links to go to the new URL by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/19
* Fleshing out the docker instructions by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/20
* Add docker install link by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/21
* Fix the error on the cron admin page by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/23
* Getting ready to make a release package by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/25
* Fix several PHP deprecation notices in the admin panel. by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/30
* Fixing build to include user_images, also seperating extra template sets by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/35
* Update version number to 20.0.0 in software by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/38
* Add a link to the license agreement on github on install and upgrade pages by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/39
* Update old copyright notices to say distributed freely under MIT license by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/40
* Add link to github releases in the list of changelogs by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/43
* Fix the PHP version it shows when upgrading and the version check fai… by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/46
* Admin login re-skin for GeoCore CE by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/47
* Make links during setup include the port to work better in local dock… by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/48
* Fix SQL query error in storefront cron, trying to pass a query param … by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/51
* Going through results of cs-check and cleaning up a bunch of files by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/52
* Update ADODB to latest version, and switch to using composer by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/54
* Fix up lint issues on the previously encoded files, and remove some deprecated methods by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/69
* Vscode xdebug settings that work with docker by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/70

## New Contributors
* @jonyo made their first contribution in https://github.com/geodesicsolutions-community/geocore-community/pull/7

**Full Changelog**: https://github.com/geodesicsolutions-community/geocore-community/compare/v0.0.0...v20.0.0-beta.3

<hr /><em>This discussion was created from the release <a href='https://github.com/geodesicsolutions-community/geocore-community/releases/tag/v20.0.0-beta.3'>v20.0.0-beta.3</a>.</em>

**Comment by vicos59** [comment]
I will update my Beta-2 instance and let you know if I find anything new. If not tonight, tomorrow for sure.

**Comment by vicos59** [comment]
A few folders have those pesky Mac files, **.DS_Store**.

upgrade/ and vendor/ for starters. I just spot checked a few. Not all folders have it.

**Comment by vicos59** [comment]
Beta2-->Beta3 upgrade was successful.

Notes:

1) I forced the Min version back to 7.2

2) Got tired of unzipping and FTPing up all of those source files. So, I just FTP'd the zip file and extracted it right on the server. Much faster.

3) My server runs the Apache process as the account user, not nobody. So, there is no need to worry about setting certain directories to the 777 like you used to have to do. I assume most modern web servers work this way. Need to review our docs in regard to this.

**Comment by vicos59** [comment]
@jonyo Have you decided where you are going to draw the line as far as releasing the first Production Code?

I am asking for selfish reasons. My monthly server lease renews near the end of the month, so if the production copy is released in the not too distant future I stand a chance of updating my v7.5.1 instance and moving it to the new server so I can shutdown the old one. Just trying to do some planning.

**Reply by jonyo** [reply]
> Have you decided where you are going to draw the line as far as releasing the first Production Code?

I haven't thought that deeply about it TBH.  So thinking about it now - given that it is now open source and we make progress only as personal time allows - no hard release dates.

Probably a good metric will be that the beta is out for at least a week with no blocker bugs found (and that is with at least 1-2 people, preferably more, actively testing).

> so if the production copy is released in the not too distant future

By the way, the version is just a label, there isn't anything stopping you from using it in production as long as you have confirmed it works by testing all the main areas you use for your own site.  I mean don't just use it w/o testing, but once you have tested and it works, go for it!

I remember back in the day, we had Geo Class Auctions 1.0 in "beta" release for well over a year, it had long been stable but I guess we were nervous to release a non-beta version.  Folks were using it in production no problemo.  It was after seeing this that I enacted (being an employee there at the time) the rule of thumb to ship it as stable release once a beta was out more than a period of time and no major bugs, and to avoid using beta unless it was huge changes like template rewrite or something.  Whatever it took to not get in the "beta forever" loop like gmail was back in the day.  But I digress...

**Reply by vicos59** [reply]
Understood! Given that, I'll set a target for myself of when the Facebook Login works and the PayPal has been upgraded since those are 2 key features I need.

**Reply by vicos59** [reply]
I'll also start testing the upgrade from 7.5.1 as well as the User side (vs just admin).

**Comment by vicos59** [comment]
I updated my server to include PHP 7.4 and I switched my GeoCore CE site to use 7.4. Everything seems to be working.

I switched on display PHP errors and see the following when /admin/ loads:

```
Deprecated: Methods with the same name as their class will not be constructors in a future version of PHP; addon_attention_getters_admin has a deprecated constructor in /home/[snip]/public_html/addons/attention_getters/admin.php on line 7

Deprecated: Methods with the same name as their class will not be constructors in a future version of PHP; addon_signs_flyers_admin has a deprecated constructor in /home/[snip]/public_html/addons/signs_flyers/admin.php on line 7
```

These are documented in Issue #75 

Friday, I will revisit all of the things I reported today as issues that were probably just related to my running PHP 7.2 vs 7.4 and get everything straightened out.

**Reply by jonyo** [reply]
That means your PHP 7.4 config is not set up with mysqli - which means it probably doesn't have mysql turned on.  Check your configs for 7.4 and if it has something for mysql, turn it on.

**Reply by vicos59** [reply]
When I installed 7.4 I thought I had the switch on to install the same extensions as I had for 7.2. But nope. I figured it out.

**Comment by vicos59** [comment]
Is there a place to put new feature requests? other than issues? I see someone posted one over on website section a few months ago.

https://github.com/geodesicsolutions-community/website/issues/4

Not exactly sure what he is asking here, but just thinking in general about differentiating new features vs bugs.

**Reply by jonyo** [reply]
Yeah "issues" is where to put feature requests as well, there is even a label for them `Enhancement`.

**Reply by vicos59** [reply]
Is there anyway to move that issue from the website section? If not, I can repost it here.

**Reply by jonyo** [reply]
I looked and did not see a way.

**Reply by vicos59** [reply]
Reposted as #80
